packages/account/src/BiconomySmartAccountV2.ts

```diff
@@ -1043,16 +1043,27 @@ export class BiconomySmartAccountV2 extends BaseSmartContractAccount {
     // for this Smart Account current validation module dummy signature will be used to estimate gas
     userOp.signature = signature;
 
-    // Note: Can change the default behaviour of calling estimations using bundler/local
-    userOp = await this.estimateUserOpGas(userOp, buildUseropDto?.stateOverrideSet);
+    if (
+      buildUseropDto?.paymasterServiceData &&
+      buildUseropDto?.paymasterServiceData.mode === PaymasterMode.SPONSORED &&
+      this.paymaster instanceof BiconomyPaymaster
+    ) {
+      const gasFeeValues = await this.bundler?.getGasFeeValues();
 
-    if (buildUseropDto?.paymasterServiceData) {
-      userOp = await this.getPaymasterUserOp(userOp, buildUseropDto.paymasterServiceData);
-    }
+      // populate gasfee values and make a call to paymaster
+      userOp.maxFeePerGas = gasFeeValues?.maxFeePerGas as Hex;
+      userOp.maxPriorityFeePerGas = gasFeeValues?.maxPriorityFeePerGas as Hex;
 
-    Logger.log("UserOp after estimation ", userOp);
+      userOp = await this.getPaymasterUserOp(userOp, buildUseropDto.paymasterServiceData);
+      return userOp;
```

Comment:
Should test the cases
- Developer pass the mode as sponsored but didn't added it on dashboard.
- Added the address but haven't whitelisted the method.

As there is no try catch what is it returning, half user op or error.

Reply:
Right, will add a try catch and throw and error

Reply:
@AmanRaj1608 actually, this code will throw if method is not whitelisted or if anything else is wrong with paymaster. No need for a try catch. Added a test which checks that the sendTransaction throws an error if a method is not whitelisted.